tortoise/replay/replay_test.go

```diff
@@ -0,0 +1,73 @@
+package replay
+
+import (
+	"context"
+	"flag"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/config"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"github.com/spacemeshos/go-spacemesh/sql"
+	"github.com/spacemeshos/go-spacemesh/timesync"
+	"github.com/spacemeshos/go-spacemesh/tortoise"
+)
+
+var (
+	dbpath = flag.String("dbpath", "", "path to database")
+	level  = zap.LevelFlag("level", zapcore.InfoLevel, "set log level")
+)
+
+func TestReplayMainnet(t *testing.T) {
+	if len(*dbpath) == 0 {
+		t.Skip("set -dbpath=<PATH> to run this test")
+	}
+
+	cfg := config.MainnetConfig()
+	types.SetLayersPerEpoch(cfg.LayersPerEpoch)
+
+	log.JSONLog(true)
+	logger := log.NewWithLevel("replay", zap.NewAtomicLevelAt(*level))
+	zlog := logger.Zap()
+	opts := []tortoise.Opt{
+		tortoise.WithLogger(logger),
+		tortoise.WithConfig(cfg.Tortoise),
+	}
+
+	genesis, err := time.Parse(time.RFC3339, cfg.Genesis.GenesisTime)
+	require.NoError(t, err)
+	clock, err := timesync.NewClock(
+		timesync.WithLayerDuration(cfg.LayerDuration),
+		timesync.WithTickInterval(1*time.Second),
+		timesync.WithGenesisTime(genesis),
+		timesync.WithLogger(log.NewNop()),
+	)
+	require.NoError(t, err)
+
+	db, err := sql.Open("file:" + *dbpath)
```

Comment:
Would it make sense to open the DB in read-only mode?

Reply:
yes, added mode=ro